server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -212,6 +212,23 @@ func (e *EventChannel) HandleEvent(ctx context.Context, event *pepb.PublishBuild
 		}
 	}
 
+	// When we get the workspace status event, update the invocation in the DB
+	// so that it can be searched by its commit SHA, user name, etc. even
+	// while the invocation is still in progress.
+	if isWorkspaceStatusEvent(bazelBuildEvent) {
+		db := e.env.GetInvocationDB()
+		ti, err := db.LookupInvocation(ctx, iid)
+		if err != nil {
+			return err
+		}
+		proto := TableInvocationToProto(ti)
+		event_parser.FillInvocationFromWorkspaceStatus(bazelBuildEvent.GetWorkspaceStatus(), proto)
+		ti = tableInvocationFromProto(proto /* blobId= */, "")
+		if err := db.InsertOrUpdateInvocation(ctx, ti); err != nil {
+			return err
+		}
```

Comment:
@tylerwilliams I wasn't sure if this lookup/insert needs to be in a transaction -- is it possible for the final SQL write to happen before this one?